packages/shared/src/open-feature.ts

```diff
@@ -290,12 +290,26 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
         try {
           await provider.onClose?.();
         } catch (err) {
-          this.handleShutdownError(this._defaultProvider, err);
+          this.handleShutdownError(provider, err);
```

Comment:
This addresses a small issue where the wrong provider was being logged when an error occurred.